tools/c7n_gcp/c7n_gcp/filters/recommender.py

```diff
@@ -0,0 +1,142 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+"""
+GCP Recommender filters
+"""
+import json
+from pathlib import Path
+
+import jmespath
+
+from c7n.exceptions import PolicyValidationError
+from c7n.filters.core import Filter
+from c7n.utils import local_session, type_schema
+
+from c7n_gcp.provider import resources as gcp_resources
+
+
+RECOMMENDER_DATA_PATH = Path(__file__).parent / "recommender.json"
```

Comment:
This _seems_ safe, wondering if there are any install/import scenarios where `__file__` will bite us though :thinking:

Reply:
good question...

our current distribution channels (wheel and docker) are fine. poetry automatically includes data files if their in package and in git. 

the one place that this makes me wonder about though is our own packaging for server less runtimes which I think is py file specific.

Reply:
added a test to verify the data file is in the server less zip, fwiw it was.

Reply:
also went down the rabbit hole of finding out about pkgresources (deprecated for importlib.resources as of py3.7) and zip_safe (also deprecated with the advent of wheels),  I'm not seeing any issues with this approach except if we were to try and do a static builder, but that's enough work on its own, that the change here shouldn't be meaningful.

https://setuptools.pypa.io/en/latest/deprecated/zip_safe.html
https://importlib-resources.readthedocs.io/en/latest/using.html
https://setuptools.pypa.io/en/latest/pkg_resources.html

Reply:
I recall you were checking out pyoxidizer for c7n, if you think that's still something viable, I'm happy to switch this to importlib.resources else, I think we can leave as is.

Reply:
Pyoxidizer experimentation is probably what triggered the `__file__` question. But realistically `__file__` seems to work fine here, and avoids the shifting sands of pkg_resources vs. pkgutil vs. importlib_resources vs. importlib.resources... ship it! :sunglasses: